gnmidiff/setrequest.go

```diff
@@ -17,13 +17,183 @@ package gnmidiff
+		var paths []string
+		for path := range deletePaths {
+			paths = append(paths, path)
+
+		}
+		sort.Strings(paths)
+		for _, path := range paths {
+			b.WriteString(fmt.Sprintf("%c %s: deleted\n", symbol, path))
+		}
+	}
+
+	writeUpdates := func(updates map[string]interface{}, symbol rune) {
+		var paths []string
+		for path := range updates {
+			paths = append(paths, path)
+
+		}
+		sort.Strings(paths)
+		for _, path := range paths {
+			b.WriteString(fmt.Sprintf("%c %s: %v\n", symbol, path, formatJSONValue(updates[path])))
+		}
+	}
+
+	b.WriteString("SetRequestIntentDiff(-A, +B):\n")
+	b.WriteString("-------- deletes --------\n")
+	if f.Full {
+		writeDeletes(diff.CommonDeletes, ' ')
+	}
+	writeDeletes(diff.AOnlyDeletes, '-')
+	writeDeletes(diff.BOnlyDeletes, '+')
+	b.WriteString("-------- updates --------\n")
+	if f.Full {
+		writeUpdates(diff.CommonUpdates, ' ')
+	}
+	writeUpdates(diff.AOnlyUpdates, '-')
+	writeUpdates(diff.BOnlyUpdates, '+')
+	var paths []string
+	for path := range diff.MismatchedUpdates {
+		paths = append(paths, path)
+
+	}
+	sort.Strings(paths)
+	for _, path := range paths {
+		mismatch := diff.MismatchedUpdates[path]
+		b.WriteString(fmt.Sprintf("m %s:\n  - %v\n  + %v\n", path, formatJSONValue(mismatch.A), formatJSONValue(mismatch.B)))
+	}
+	return b.String()
+}
+
+// DiffSetRequest returns a unique and minimal intent diff of two SetRequests.
+//
+// newSchema is intended to be provided via the function defined in generated
+// ygot code (e.g. exampleoc.Schema).
+// If newSchema is nil, then DiffSetRequest will make the following assumption:
+// - Any JSON value in the input SetRequest MUST conform to the OpenConfig
+// YANG style guidelines. See the following for checking compliance.
+// * https://github.com/openconfig/oc-pyang
+// * https://github.com/openconfig/public/blob/master/doc/openconfig_style_guide.md
+//
+// Currently, support is only for SetRequests without any delete paths, and
+// replace and updates that don't have conflicting leaf values. If not
+// supported, then an error will be returned.
+func DiffSetRequest(a *gpb.SetRequest, b *gpb.SetRequest, newSchemaFn func() (*ytypes.Schema, error)) (SetRequestIntentDiff, error) {
+	intentA, err := minimalSetRequestIntent(a, newSchemaFn)
+	if err != nil {
+		return SetRequestIntentDiff{}, fmt.Errorf("DiffSetRequest on a: %v", err)
+	}
+	intentB, err := minimalSetRequestIntent(b, newSchemaFn)
+	if err != nil {
+		return SetRequestIntentDiff{}, fmt.Errorf("DiffSetRequest on b: %v", err)
+	}
+	diff := SetRequestIntentDiff{
+		AOnlyDeletes:      map[string]struct{}{},
+		BOnlyDeletes:      map[string]struct{}{},
+		CommonDeletes:     map[string]struct{}{},
+		AOnlyUpdates:      map[string]interface{}{},
+		BOnlyUpdates:      map[string]interface{}{},
+		CommonUpdates:     map[string]interface{}{},
+		MismatchedUpdates: map[string]MismatchedUpdate{},
+	}
+	for pathA := range intentA.Deletes {
```

Comment:
I think you can replace lines 166 through 177 with:

```
for path := range intentA.Deletes {
        if _, ok := intentB.Deletes[path]; ok {
                delete(intentA.Deletes, path)
                delete(intentB.Deletes, path)
                diff.CommonDeletes[path] = struct{}{}
        }
}
```
And then `intentA.Deletes` is your `diff.AOnlyDeletes` and same for B.

Reply:
Done.